src/backend/lib/utils.js

```diff
@@ -64,16 +111,42 @@ function installerCompare(a, b) {
 function isUpgradeCandidate({ deviceInfo, installer, currentApkInfo }) {
   // Filter out the current APK
   if (installer.hash === currentApkInfo.hash) return false;
+  // Check for 0 or less APK size
+  if (installer.size <= 0) return false;
   // TODO: 64 bit devices can run 32 bit APKs
   const isSupportedAbi = !!installer.arch.find(arch =>
     deviceInfo.supportedAbis.includes(arch)
   );
   if (!isSupportedAbi) return false;
+  log(
+    "installer.minSdkVersion > deviceInfo.sdkVersion",
+    installer.minSdkVersion > deviceInfo.sdkVersion
+  );
   if (installer.minSdkVersion > deviceInfo.sdkVersion) return false;
+  log(
+    "installer.applicationId !== currentApkInfo.applicationId",
+    installer.applicationId !== currentApkInfo.applicationId
+  );
   if (installer.applicationId !== currentApkInfo.applicationId) return false;
+  log(
+    `isUpgradeCandidate ~ installer.platform !== "android"`,
+    installer.platform !== "android"
+  );
   if (installer.platform !== "android") return false;
+  log(
+    "installer.versionCode < currentApkInfo.versionCode",
+    installer.versionCode < currentApkInfo.versionCode
+  );
   if (installer.versionCode < currentApkInfo.versionCode) return false;
```

Comment:
```suggestion
  if (installer.versionCode <= currentApkInfo.versionCode) return false;
```

We need to check `<=` not `<` to avoid recursive updates or internal builds or RCs.